### PR TITLE
Food data

### PR DESCRIPTION
**Pull request by cheburan**
Implementation of the getFoodData API endpoint
- food. controller for an `entry` 
- food-data service
- DB models related to: 
     - food Locale
     - food portionSize Methods and params
     - categories portionSize Methods and param
     - associated Foods
     - categories 
     - brands ...

### CODE REVIEW

File: api-server/src/db/models/foods/category-portion-size-method.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    allowNull: false,
+    type: DataType.FLOAT(17),
+  })
+  public conversionFactor!: number;
+
+  @BelongsTo(() => Category, 'categoryCode')
+  public category?: Category;
+
+  @HasMany(() => CategoryPortionSizeMethodParameter, 'portionSizeMethodId')
+  public parameters!: CategoryPortionSizeMethodParameter[];
```
Comment (lukashroch) [inline api-server/src/db/models/foods/category-portion-size-method.ts:68]: assertion type? Associations can be undefined if not loaded by query.

File: api-server/src/db/models/foods/portion-size-method.ts [hunk 2]
```diff
@@ -46,12 +49,13 @@ export default class PortionSizeMethod extends BaseModel {
...
+    type: DataType.FLOAT(17),
   })
   public conversionFactor!: number;
 
   @BelongsTo(() => FoodLocal, 'foodLocalId')
   public foodLocal?: FoodLocal;
 
   @HasMany(() => PortionSizeMethodParameter, 'portionSizeMethodId')
-  public parameters?: PortionSizeMethodParameter[];
+  public parameters!: PortionSizeMethodParameter[];
```
Comment (lukashroch) [inline api-server/src/db/models/foods/portion-size-method.ts:60]: same as above?

File: api-server/src/http/controllers/food.controller.ts [hunk 3]
```diff
@@ -1,49 +1,80 @@
...
-type FoodData = {
-  [key: string]: any; // FIXME: use more specific type
-};
-
 export type FoodController = Controller<
   'entry' | 'entryWithSource' | 'brands' | 'associatedFoods' | 'composition'
 >;
 
-export default (): FoodController => {
+export default ({ foodDataService }: IoC): FoodController => {
```
Comment (lukashroch) [inline api-server/src/http/controllers/food.controller.ts:10]: I would type it: `Pick<IoC, 'foodDataService'>` to allow it testable if needed.

File: common/src/types/http/foods/food-data.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
+export type FoodDataEntryResponse = FoodLocalResponse & FoodAttributeResponse & FoodAdditionalDataResponse & CaloriesPer100gResponse
+
+export type FoodLocalResponse = {
+	code: string;
+	localDescription: string;
+	portionSizeMethods: PortionSizeMethodsResponse[] | [];
```
Comment (lukashroch) [inline common/src/types/http/foods/food-data.ts:6]: does this provide any type advantage over just `PortionSizeMethodsResponse[]`? <-- this allows empty array as well.

File: api-server/src/db/models/foods/category-portion-size-method.ts (hunk 1 above)
Comment (cheburan) [reply api-server/src/db/models/foods/category-portion-size-method.ts]: > assertion typo? Associations can be undefined if not loaded by query.

yep, addressed

File: api-server/src/db/models/foods/portion-size-method.ts (hunk 2 above)
Comment (cheburan) [reply api-server/src/db/models/foods/portion-size-method.ts]: > same as above?

cheers, yep changed

File: api-server/src/http/controllers/food.controller.ts (hunk 3 above)
Comment (cheburan) [reply api-server/src/http/controllers/food.controller.ts]: yep, thanks for the suggestion.

File: common/src/types/http/foods/food-data.ts (hunk 4 above)
Comment (cheburan) [reply common/src/types/http/foods/food-data.ts]: yeah, it is actually redundant at that point, nice spotting it!

**Review by lukashroch — APPROVED** [review]

**Pull request merged**